client/src/pages/Product/Product.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { CustomButton } from '../../components/common/CustomButton/CustomButton';
+import { InfoMessage } from '../../components/common/InfoMessage/InfoMessage';
+import { Preloader } from '../../components/common/Preloader/Preloader';
+import { ComponentDto } from '../../dtos/components/component.dto';
+import { useAppDispatch, useAppSelector } from '../../hooks/redux';
+import { API_URL } from '../../http/http';
+import { getProductById } from '../../stateManager/actionCreators/products';
+import { setCurrentProduct } from '../../stateManager/slices/productsSlice';
+import styles from './Product.module.scss';
+
+interface IProps {}
+
+export const Product: React.FC<IProps> = ({}) => {
+    const dispatch = useAppDispatch();
+    const isFetching = useAppSelector((state) => state.products.isFetching);
+    const product = useAppSelector((state) => state.products.currentProduct);
+    const { productId } = useParams();
+
+    useEffect(() => {
+        dispatch(getProductById(Number(productId)));
+
+        return () => {
+            dispatch(setCurrentProduct(null));
+        };
+    }, []);
+
+    const renderComponents = () => {
+        return (
+            product?.components &&
```

Comment:
Notice: It is more common practice to not concat the rendering with boolean values, but to create exit condition instead.

So you can do something like 
```typescript
 if (!product?.components || product.components.length === 0) { return null };
          return (
              <div className={styles.productComponents}>
                  <h4 className={styles.componentsTitle}>Ingredients</h4>
                  <div className={styles.componentsBody}>
                      {product.components.map(renderComponent)}
                  </div>
              </div>
          );

```